v3-sdk/routing/src/example/Example.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import './Example.css'
+import { Environment, CurrentConfig } from '../config'
+import { getCurrencyBalance } from '../libs/wallet'
+import {
+  connectBrowserExtensionWallet,
+  getProvider,
+  getWalletAddress,
+  TransactionState,
+} from '../libs/providers'
+import { executeRoute, generateRoute } from '../routing'
+import { SwapRoute } from '@uniswap/smart-order-router'
+
+const useOnBlockUpdated = (callback: (blockNumber: number) => void) => {
+  useEffect(() => {
+    const subscription = getProvider()?.on('block', callback)
+    return () => {
+      subscription?.removeAllListeners()
+    }
+  })
+}
+
+const Example = () => {
+  const [tokenInBalance, setTokenInBalance] = useState<string>()
+  const [tokenOutBalance, setTokenOutBalance] = useState<string>()
+  const [txState, setTxState] = useState<TransactionState>(TransactionState.New)
+  const [blockNumber, setBlockNumber] = useState<number>(0)
+
+  const [route, setRoute] = useState<SwapRoute | null>(null)
+
+  // Listen for new blocks and update the wallet
+  useOnBlockUpdated(async (blockNumber: number) => {
+    refreshBalances()
+    setBlockNumber(blockNumber)
+  })
+
+  // Update wallet state given a block number
+  const refreshBalances = useCallback(async () => {
+    const provider = getProvider()
+    const address = getWalletAddress()
+    if (address && provider) {
+      setTokenInBalance(
+        await getCurrencyBalance(provider, address, CurrentConfig.currencies.in)
+      )
+      setTokenOutBalance(
+        await getCurrencyBalance(
+          provider,
+          address,
+          CurrentConfig.currencies.out
+        )
+      )
+    }
+  }, [])
+
+  // Event Handlers
+
+  const onConnectWallet = useCallback(async () => {
+    if (await connectBrowserExtensionWallet()) {
+      refreshBalances()
+    }
+  }, [refreshBalances])
+
+  const onCreateRoute = useCallback(async () => {
+    setRoute(await generateRoute())
+  }, [])
+
+  const onSwap = useCallback(async () => {
```

Comment:
Should we pass the `route` here as an argument to the function? Makes is more clear when we'll change this.

Reply:
What will this change to? In this follows the same pattern as the trading example where we use the React state to handle the inputs etc

Reply:
I guess I haven't reviewed the trade PR yet - so this is new for me. However, pulling context variables from within the function is less maintainable/readable that having them explicitly called out in the arg list